### DISCUSSION TITLE
How to use the aac encoding format when integrating shaiport sync 4.3.2 into Linux devices with arm64 architecture?

### DISCUSSION THREAD

**Discussion by Biao7** [mikebrady/shairport-sync#1806] [Q&A]
Hi mikebrady，
I have a question：
I integrated sharport sync 4.3.2 in the Linux system, which supports airplay 2 functionality. I am using an iPhone to connect to a Linux device and can connect normally, but the audio encoding transmitted is default to using Alac. How can I have my phone send audio data encoded with AAC? Can we modify the code of sharport sync to force the use of aac encoding format for data processing during connection negotiation?
Thanks

**Comment by mikebrady** [comment]
Thanks for the question.

With Shairport Sync, AirPlay 2 operates in one of two modes: a "Realtime" mode in which it uses ALAC encoding, and a "Buffered" mode in which is uses AAC. The mode is chosen by the source. For instance, Spotify on iOS (still) uses Realtime/ALAC and the Apple Music app on iOS uses Buffered/AAC. As far as is known, this choice of mode can not be influenced by the player.

(As usual, please bear in mind that the full specification of AirPlay 2 has not been published, and so we only have empirical knowledge.)

**Reply by Biao7** [reply]
Thanks for your answer.

From your description above, what decoding method to use is solely determined by the phone. Is this understanding correct? I originally guessed that during the process of establishing an AirPlay2 connection between the phone and the device, there would be a handshake process to negotiate which encoding and decoding method to use for data transmission. However, I did not find any relevant code after reading the code. 

In summary, is it correct for sharport sync not to participate in the selection of encoding and decoding methods? It is only responsible for identifying the type of audio data received and decoding it using the corresponding method. Is this understanding correct?

Thanks

**Reply by mikebrady** [reply]
Yes, your understanding is correct -- that is what seems to be the case.

**Reply by Biao7** [reply]
Ok!Thanks for your answer.
